python/dgl/dataloading/base.py

```diff
@@ -271,9 +410,171 @@ def sample(self, g, seed_edges):
             g, seed_edges, exclude, self.reverse_eids, self.reverse_etypes,
             self.output_device)
 
-        input_nodes, _, blocks = self.block_sampler.sample_blocks(g, seed_nodes, exclude_eids)
+        input_nodes, _, blocks = self.sampler.sample(g, seed_nodes, exclude_eids)
 
         if self.negative_sampler is None:
             return self.assign_lazy_features((input_nodes, pair_graph, blocks))
         else:
             return self.assign_lazy_features((input_nodes, pair_graph, neg_graph, blocks))
+
+def as_edge_prediction_sampler(
+        sampler, exclude=None, reverse_eids=None, reverse_etypes=None, negative_sampler=None,
+        prefetch_labels=None):
```

Comment:
Hi @zjost , I wonder if you could take a look at this function and the example [here](https://github.com/dmlc/dgl/blob/02367940aaa7cef129ff95e6e5f80be2ad0ce304/examples/pytorch/graphsage/link_pred.py). Do you think if there are some other more intuitive names? I feel `as_edge_prediction_sampler` seems to be a bit over claim, i.e., there may be some other ways to create an edge-wise sampler from a node-wise sampler. I am thinking of `TwoTowerEdgeSampler` but don't know if it looks straightforward to users.

Reply:
Also cc @mufeili